ncm-grub/src/main/perl/grub.pm

```diff
@@ -74,6 +77,66 @@ sub grubbyArgsOptions {
 
 }
 
+sub Configure_password {
+    my ($self, $config, $editor) = @_;
+
+    my $password = undef;
+    my $passwordpath = $self->prefix . "/password";
+
+    # if passwords have not been explicitly enabled or disabled then do nothing.
+    return unless ($config->elementExists($passwordpath));
+    my $tree = $config->getElement($passwordpath)->getTree();
+
+    if (!defined($tree->{enabled})) {
+        return;
+    } elsif (!$tree->{enabled}) {
+        $editor->remove_lines(qr/^password\s+/, '');
+        return;
+    }
+
+    if (my $passwordfile = $tree->{file}) {
+        my $fileuser = $tree->{file_user};
+        if (! -R $passwordfile) {
+            $self->error("grub password file $passwordfile does not exist or is not readable.");
+            return;
+        }
+        my $pf_fh = CAF::FileEditor->new($passwordfile, log => $self);
+        $pf_fh->cancel();
+        for my $line (split(/\n/, "$pf_fh")) {
```

Comment:
`for` -> `foreach` (it doens't actaully do anything different, but we use `foreach` in all the other code)